statistics/handle/handle.go

```diff
@@ -1408,38 +1417,51 @@ func (sr *statsReader) isHistory() bool {
 	return sr.snapshot > 0
 }
 
-func (h *Handle) getStatsReader(snapshot uint64) (reader *statsReader, err error) {
+func (h *Handle) getGlobalStatsReader(snapshot uint64) (reader *statsReader, err error) {
+	h.mu.Lock()
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("getGlobalStatsReader panic %v", r)
+		}
+		if err != nil {
```

Comment:
Why doesn't mu unlock if err is not null？ Will it cause an deadlock？

Reply:
The normal unlock is in releaseGlobalStatsReader()